save-demo-agent/src/nativeMain/kotlin/com/saveourtool/save/demo/agent/utils/HttpUtils.kt

```diff
@@ -0,0 +1,95 @@
+/**
+ * Utils to send http requests
+ */
+
+package com.saveourtool.save.demo.agent.utils
+
+import com.saveourtool.save.core.logging.logDebug
+import com.saveourtool.save.core.logging.logError
+import com.saveourtool.save.core.logging.logInfo
+import com.saveourtool.save.core.logging.logWarn
+import com.saveourtool.save.core.utils.runIf
+import com.saveourtool.save.utils.failureOrNotOk
+import com.saveourtool.save.utils.fs
+import com.saveourtool.save.utils.notOk
+import io.ktor.client.*
+import io.ktor.client.call.*
+import io.ktor.client.engine.cio.*
+import io.ktor.client.request.*
+import io.ktor.client.statement.*
+import io.ktor.client.utils.*
+import io.ktor.http.*
+import io.ktor.utils.io.*
+import io.ktor.utils.io.core.*
+import okio.Path
+import okio.buffer
+import okio.use
+import kotlin.native.concurrent.AtomicLong
+
+private val httpClient = HttpClient(CIO)
+
+private suspend fun HttpClient.download(url: String, file: Path): Result<HttpResponse> = runCatching {
+    prepareGet {
+        url(url)
+        contentType(ContentType.Application.Json)
+        accept(ContentType.Application.OctetStream)
+    }
+        .execute { httpResponse ->
+            if (httpResponse.status.isSuccess()) {
+                val channel: ByteReadChannel = httpResponse.body()
+                val totalBytes = AtomicLong(0L)
+                while (!channel.isClosedForRead) {
+                    val packet = channel.readRemaining(DEFAULT_HTTP_BUFFER_SIZE.toLong())
+                    while (!packet.isEmpty) {
+                        val bytes = packet.readBytes()
+                        val hasAlreadyAnyData = totalBytes.value > 0
+                        fs.appendingSink(file, mustExist = hasAlreadyAnyData).buffer().use { it.write(bytes) }
+                        totalBytes.addAndGet(bytes.size.toLong())
+                        logDebug("Received ${bytes.size} ($totalBytes) bytes out of ${httpResponse.contentLength()} bytes from ${httpResponse.request.url}")
+                    }
+                }
+                if (totalBytes.value == 0L) {
+                    error("Downloaded a file from $url but content is empty")
+                }
+            } else {
+                logWarn("Skipping downloading as request is not a success")
```

Comment:
Please log the status code and status line (e. g.: `HTTP 418 I'm a teapot`.

Reply:
Will do it